pxf/pxf-jdbc/src/main/java/org/apache/hawq/pxf/plugins/jdbc/JdbcPartitionFragmenter.java

```diff
@@ -111,30 +110,40 @@ public JdbcPartitionFragmenter(InputData inConf) throws UserDataException {
             partitionBy = inConf.getUserProperty("PARTITION_BY").split(":");
             partitionColumn = partitionBy[0];
             partitionType = PartitionType.getType(partitionBy[1]);
-        } catch (IllegalArgumentException e1) {
+        } catch (IllegalArgumentException | ArrayIndexOutOfBoundsException e1) {
             throw new UserDataException("The parameter 'PARTITION_BY' invalid, the pattern is 'column_name:date|int|enum'");
         }
 
+        //parse and validate parameter-RANGE
         try {
-            range = inConf.getUserProperty("RANGE").split(":");
+            if (inConf.getUserProperty("RANGE") != null) {
+                range = inConf.getUserProperty("RANGE").split(":");
+                if (range.length == 1 && partitionType != PartitionType.ENUM)
+                    throw new UserDataException("The parameter 'RANGE' does not specify '[:end_value]'");

```

Comment:
If we want to support open-ended range on fragmenter phase - we might have the last fragment having more data than rest of fragments, which is acceptable and very user-friendly so +1 for having open-ended range.